torchdata/datapipes/iter/util/combining.py

```diff
@@ -115,13 +115,12 @@ def __iter__(self) -> Iterator:
                 else:
                     yield res
         finally:
-            for remaining in ref_it:
-                janitor(remaining)
-
+            del ref_it
```

Comment:
Question: IIUC, we are trusting there is nothing needs to be cleaned for `ref_it` or `ref_dp` will handle it?

Reply:
noob question: `del ref_it` will trigger `ref_it.__del__()` right? -- which seems to doing the cleaning? ~

Reply:
> IIUC, we are trusting there is nothing needs to be cleaned for `ref_it` or `ref_dp` will handle it?

> does that mean we believe `del ref_it` will do the clean up?

We have to. If the prior DataPipe is an infinite one like `cycle(None)`, it means we will never reach the end of `ref_it`. It's more like a safe guard to explicitly remove `ref_it` from the scope. 
And, this is the reason that all generators that yields opened file handles are required to using `finally` invoke `file.close` to guarantee the file is closed when `generator` object is out of scope.